contents/docs/self-host/configure/migrate-postgres-clickhouse.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Migrate from Postgres to ClickHouse
+sidebar: Docs
+showTitle: true
+---
+
+> This documentation is in **Beta**. Please provide feedback via the [PostHog Communty Slack workspace](/slack) or [via a GitHub issue](https://github.com/PostHog/posthog.com/issues).
+
+**PostHog backed by Postgres is now deprecated**. We will continue to provide support to Postgres backed installs but we strongly encourage you to migrate to PostHog backed by ClickHouse for improved performance, to receive new features, and continued support in the future.
+
+## Before you begin
+
+TODOs:
+- Is there a minimum version for PostHog + Postgres and a minimum version for PostHog + ClickHouse? For example, does the Replicator plugin have a minimum PostHog version to work within?
+
+In order to migrate you must have the following in place:
+
+1. The PostHog + Postgres installation
+2. A new PostHog + ClickHouse installation
+
+## Export
+
+TODOs:
+- Are these the correct tables to dump?
+- Should we be using `--data-only` to only dump the data and not the schema creation?
```

Comment:
@timgl @mariusandra & anyone else - can we get confirmation on the tables that needs to be dumped here, please?

Reply:
I would do everything except `posthog_person`, `posthog_persondistinctid`, `posthog_event`, `posthog_pluginlogentry` and `posthog_sessionrecordingevent`

`--data-only` probably makes sense as users will have most likely already migrated.

Reply:
+1 for `--data-only` and adding a small note about migrating first

Reply:
Ah sorry so migration is used in two contexts here - "migrating" the data and "migrating" the database

We should tell people to migrate their _database_ before running this guide and only export the data.

Reply:
My understanding is that we were providing details on how to migrate their Postgres DB by providing `pg_dump` and `pg_restore` instructions?

Reply:
FYI, I was experimenting on our staging environment and tried this with `--data-only`. The new PH-Clickhouse instance was clean; I hadn't gone through the preflight steps.

It didn't seem to migrate our project correctly. All I could see after `psql [...] < migration.sql` was a "PostHog demo project" that couldn't be renamed.